src/components/Map.jsx

```diff
@@ -1,27 +1,60 @@
+import T from "prop-types";
 import MapGl, { Layer, Source } from "react-map-gl/maplibre";
 import "maplibre-gl/dist/maplibre-gl.css";
+import bbox from "@turf/bbox";
+import { AttributionControl } from "react-map-gl";
 
+function Map({ artifact, mini }){
+  const isStatic = mini; // if mini is true, the map is static
+  let viewState = {
+    longitude: 0,
+    latitude: 0,
+    zoom: 0
+  };
+
+  if (artifact) {
+    const bounds = bbox(artifact);
+    viewState = {
+      longitude: (bounds[0] + bounds[2]) / 2,
+      latitude: (bounds[1] + bounds[3]) / 2,
+      zoom: 12
+    };
+  }
 
-function Map() {
   return (
     <MapGl
-      initialViewState={{
-        longitude: 0,
-        latitude: 0,
-        zoom: 0
-      }}
+      initialViewState={viewState}
+      touchPitch={!isStatic}
+      touchZoom={!isStatic}
+      dragRotate={!isStatic}
+      scrollZoom={!isStatic}
+      attributionControl={false}
     >
       <Source
         id="background"
         type="raster"
         tiles={["https://tile.openstreetmap.org/{z}/{x}/{y}.png"]}
         tileSize={256}
-        attribution="Background tiles: © <a href='https://www.openstreetmap.org/copyright'>OpenStreetMap contributors</a>"
       >
         <Layer id="background-tiles" type="raster" />
       </Source>
+      {artifact && (
+        <Source
+          id="artifact"
+          type="geojson"
+          data={artifact}
+        >
+          <Layer id="artifact-layer" type="line" paint={{ "line-color": "#ff0000", "line-width": 2 }} />
+        </Source>
+      )}
```

Comment:
I would design this a bit more flexible so we can use the map component in different ways. If we do it this way we can only add one layer at a time but we might want more, for example displaying all the artefacts on the big map together.

I'd move the `Source` and `Layer` initialisation to the component that renders the map, in this case the `ContextLayer` and `DistAlertsTool` components, and change the `Map` component so it accepts a `children` prop that we render here.